command/server.go

```diff
@@ -446,9 +446,35 @@ func (c *ServerCommand) parseConfig() (*server.Config, []configutil.ConfigError,
 		c.UI.Warn("WARNING: Entropy Augmentation is not supported in FIPS 140-2 Inside mode; disabling from server configuration!\n")
 		config.Entropy = nil
 	}
-
 	entCheckRequestLimiter(c, config)
 
+	if config != nil {
+		fmt.Printf("SharedConfig %+v\n", config.SharedConfig)
+	} else {
+		fmt.Printf("SharedConfig missing: Config %+v\n", config)
+	}
+	// ensure that the DisableMlock key is explicitly set
+	if config != nil && config.SharedConfig != nil &&
+		// DisableMlock key has been found and thus explicitly set
+		(!strutil.StrListContainsCaseInsensitive(config.SharedConfig.FoundKeys, "DisableMlock") ||
+			// mlock is disabled and hence has been explicitly set
+			config.SharedConfig.DisableMlock == true) {
```

Comment:
This check is because in our Docker cluster tests, `FoundKeys` are not being added to the `SharedConfig`. I am not sure if this is intentional yet, but am looking into it.